pkg/handler/commit.go

```diff
@@ -101,3 +102,87 @@ func (h *commitHandler) GetCommitByMessage(w http.ResponseWriter, r *http.Reques
 
 	_ = json.NewEncoder(w).Encode(resp)
 }
+
+func (h *commitHandler) CommitReleased(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	// Get request data from query params
+	request, errMessage := GetCommitReleasedRequest(r)
+	if errMessage != "" {
+		http.Error(w, errMessage, http.StatusBadRequest)
+		return
+	}
+
+	branchExists, err := checkIfBranchExists(request.Owner, request.Repository, request.ReleaseBranch)
+	if err != nil {
+		http.Error(w, "Error checking if branch exists: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if !branchExists {
+		http.Error(w, "Branch does not exist in the repository", http.StatusBadRequest)
+		return
+	}
+
+	baseUrl := fmt.Sprintf("https://api.github.com/repos/%s/%s/commits?sha=%s&per_page=100&page=", request.Owner, request.Repository, request.ReleaseBranch)
+	method := "GET"
+
+	req, err := http.NewRequest(method, baseUrl, nil)
+	if err != nil {
+		http.Error(w, "Error generating new request: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	commitReleased := false
+
+	for page_number := 1; ; page_number++ {
+		url := baseUrl + strconv.Itoa(page_number)
+		u, err := urlpkg.Parse(url)
+		if err != nil {
+			http.Error(w, "Error generating new URL: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		req.URL = u
+
+		// Define a variable of type []Commit to store the data
+		var commits []model.CommitData
+
+		client := &http.Client{}
```

Comment:
i think we can initialize the client outside the loop to avoid creating a new one for each iteration of the loop

_i dont have a lot of experience working with golang so i might be wrong_ 😅

Reply:
Makes sense, I didn't notice that! I have initialized it outside the for loop and tested the flow as well. Thanks